test/gui/shared/steps/sync_context.py

```diff
@@ -296,3 +296,23 @@ def step(context, should_or_should_not):
 @When('the user unchecks the "|any|" filter')
 def step(context, filter_option):
     Activity.select_not_synced_filter(filter_option)
+
+
+@Then('the following error message should appear in the client')
+def step(context):
+    expected_error_message = '\n'.join(context.multiLineText)
+
+    actual_error_message = SyncConnection.get_permission_error_message()
+
+    # wait for error message to disappear
+    squish.waitFor(
+            lambda: SyncConnection.get_permission_error_message() == "",
+            get_config("minSyncTimeout") * 1000,
+        )
```

Comment:
we can have a separate method in the page object file
```suggestion
    def wait_for_error_label(to_exist=True):
        status = squish.waitFor(
            lambda: object.exists(SyncConnection.PERMISSION_ERROR_LABEL) == to_exist,
            SyncConnection.PERMISSION_ERROR_WAIT_TIMEOUT * 1000
        )
        if status != to_exist:
            <raise error>
```
then we can use it to wait for label to exist or disappear:
`wait_for_error_label(True)`, `wait_for_error_label(False)`